Xamarin.PropertyEditing.Mac/Controls/EditorContainer.cs

```diff
@@ -63,6 +95,209 @@ public override void ViewWillMoveToSuperview (NSView newSuperview)
 			base.ViewWillMoveToSuperview (newSuperview);
 		}
 
+		public sealed override void ViewDidChangeEffectiveAppearance ()
+		{
+			base.ViewDidChangeEffectiveAppearance ();
+
+			AppearanceChanged ();
+		}
+
+		public override void Layout ()
+		{
+			base.Layout ();
+
+			FrameChanged ();
+		}
+
+		protected void AppearanceChanged ()
+		{
+			// Repaint those lines
+			foreach (CALayer variationLayer in this.variationParentLayersList) {
+				if (variationLayer is CAShapeLayer lineShape) {
+					lineShape.StrokeColor = HostResources.GetNamedColor (NamedResources.ForegroundColor).CGColor;
+				}
+			}
+
+			foreach (CALayer variationLayer in this.variationChildLayersList) {
+				if (variationLayer is CAShapeLayer lineShape) {
+					lineShape.StrokeColor = HostResources.GetNamedColor (NamedResources.ForegroundColor).CGColor;
+				}
+			}
+		}
+
+		private void UpdateVariations ()
+		{
+			ClearSubViews ();
```

Comment:
You have checks to only create if there aren't buttons, but you're erasing the buttons here, recreating them regardless.

Reply:
They need to be cleared, so have removed the checks.